### PR TITLE
[WNMGDS-1307] WHCM badges

### PR DESCRIPTION
**Pull request by zarahzachz**
## Summary

[Jira ticket](https://jira.cms.gov/browse/WNMGDS-1307)

hcm-badges

Implement a HCM query via React.

## How to test

Test [this link](http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS/1307-whcm-badges/core/storybook/?path=/story/components-badge--default) via your favorite VM.

### CODE REVIEW

File: packages/design-system/src/styles/components/_Badge.scss [hunk 1]
```diff
@@ -2,6 +2,26 @@
...
+@mixin badge-high-contrast-mode-label($label) {
+  /* stylelint-disable scss/media-feature-value-dollar-variable */
+  @media (-ms-high-contrast: active), (forced-colors: active) {
+    &::before {
+      content: $label + ': ';
+    }
+  }
+  /* stylelint-enable scss/media-feature-value-dollar-variable */
+}
+
```
Comment (ericwbailey) [inline packages/design-system/src/styles/components/_Badge.scss:24]:
```suggestion
```

File: packages/design-system/src/styles/components/_Badge.scss [hunk 2]
```diff
@@ -23,17 +45,29 @@ $badge-font-size-big: $base-font-size;
 
 .ds-c-badge--info {
   background-color: $color-primary;
+
+  @include badge-high-contrast-mode-remove-color;
+  @include badge-high-contrast-mode-label('Notice');
```
Comment (ericwbailey) [inline packages/design-system/src/styles/components/_Badge.scss:50]:
```suggestion
```

File: packages/design-system/src/styles/components/_Badge.scss [hunk 3]
```diff
@@ -23,17 +45,29 @@ $badge-font-size-big: $base-font-size;
...
 }
 
 .ds-c-badge--success {
   background-color: $color-success;
+
+  @include badge-high-contrast-mode-remove-color;
+  @include badge-high-contrast-mode-label('Success');
```
Comment (ericwbailey) [inline packages/design-system/src/styles/components/_Badge.scss:57]:
```suggestion
```

File: packages/design-system/src/styles/components/_Badge.scss [hunk 4]
```diff
@@ -23,17 +45,29 @@ $badge-font-size-big: $base-font-size;
...
 }
 
 .ds-c-badge--warn {
   background-color: $color-warn;
   color: $color-base;
+
+  @include badge-high-contrast-mode-remove-color;
+  @include badge-high-contrast-mode-label('Warning');
```
Comment (ericwbailey) [inline packages/design-system/src/styles/components/_Badge.scss:65]:
```suggestion
```

File: packages/design-system/src/styles/components/_Badge.scss [hunk 5]
```diff
@@ -23,17 +45,29 @@ $badge-font-size-big: $base-font-size;
...
 }
 
 .ds-c-badge--alert {
   background-color: $color-error;
+
+  @include badge-high-contrast-mode-remove-color;
+  @include badge-high-contrast-mode-label('Alert');
```
Comment (ericwbailey) [inline packages/design-system/src/styles/components/_Badge.scss:72]:
```suggestion
```

File: packages/design-system/src/styles/components/_Badge.scss (hunk 1 above)
Comment (scheul93) [inline packages/design-system/src/styles/components/_Badge.scss:24]: Is this still necessary if the label is now being added in markup?

**Review by ericwbailey — CHANGES_REQUESTED** [review]
I'm seeing a merge conflict. Also, I snuck in and removed the Sass related to generating CSS HCM content.

**Comment by pwolfert** [comment]
Question: Would it be a helpful simplification to always render the prefix but hide it outside of WHCM with CSS? I could see us wanting to show it for other modes, like maybe if we detect color-blindness settings?

File: packages/design-system/src/styles/components/_Badge.scss (hunk 1 above)
Comment (zarahzachz) [reply packages/design-system/src/styles/components/_Badge.scss]: No, it's not needed. I've removed all the `@mixin` style relating to label.

**Pull request closed without merging**